### ISSUE TITLE
jackson does not honor generics or json property

### ISSUE THREAD

**Issue by jigarparekh80** [stoicflame/enunciate#338]
I have common response class which have been inherited with java generics

```
public class CommonResponse<T> {
    private Integer code; 
    private String status;
    private T payload;
}

public class UserResponse extends CommonResponse<User> {

    @Override
    @JsonProperty(value="user")
    public User getPayload() {
        return super.getPayload();
    }
 }
```

jackson does not recognize User property and generates below example 

```
{
  "status" : "...",
  "payload" : { },
  "code" : 12345
}
```

and in datatype section it shows payload as object instead of User.

**Comment by stoicflame** [comment]
I acknowledge the issue, but I can't figure out what to do about it.

One option would be to document the `payload` property as if it belonged to the `UserResponse` data type, but then there would be another (duplicate) property that belongs to the `CommonResponse` data type. So under "Properties of UserResponse" you'd see a property `payload` of type "User", but then a little bit lower in the list, you'd see a section "Properties of CommonResponse" that _also_ contains a property `payload` of type "object". That seems really confusing to me.

I suppose we could "hide" the second property if it shows up in the context of the subtype, but it's hard for me to wrap my head around what that's going to do with the example requests and property ordering.

**Comment by jigarparekh80** [comment]
yes, i have tried first option but as you have suggested it results in to two property. 

problem is jsonproperty does not allow to specify type so might difficult to get type information. 
is it possible to read type value from @xmlelement? (may be workaround)

**Comment by stoicflame** [comment]
Sure, you could use `@TypeHint(User.class)` or you could use `@JsonSerialize(as=User.class)` to specify a hint, but then if there were _other_ subtypes of `CommonResponse`, they'd get documented as if their payload were of type `User`.

**Comment by stoicflame** [comment]
I spent some more time on this, but I keep running into dead ends.

I looked into the possibility of making the base class/methods abstract and overriding them, but it doesn't make any difference.

I looked into the possibility of applying `@JsonIgnore` to the base class properties, but then Jackson ignores it.

I looked into the possibility of applying `@Ignore` to the base class, but then the type parameter doesn't get resolved in the context of the subclass.

At this point, I just don't know how to accommodate this use case. Jackson looks at the superclass for property info, but has additional runtime information that Enunciate simply doesn't have at compile-time. 

For now, I'm going to close this as "will not fix", but I'm happy to re-open if anyone can propose a workable solution.
